src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.Windows.cs

```diff
@@ -406,6 +481,37 @@ internal static unsafe int WriteFileNative(SafeFileHandle handle, ReadOnlySpan<b
             }
         }
 
+        private static void GetPathForNtCreateFile(string fullPath, out ReadOnlySpan<char> prefixedAbsolutePath, out char[]? rentedArray)
+        {
+            Debug.Assert(fullPath == Path.GetFullPath(fullPath));
```

Comment:
This assert is not guaranteed to hold. The file system can be changing underneath you.